src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,99 @@
+package storage;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+import tasklist.TaskList;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+
+public class Storage {
+    private String filepath;
+    private Path path;
+    public Storage(String filepath) {
+        this.filepath = filepath;
+        this.path = Paths.get(this.filepath);
+    }
+    /**
+     * Create the directory and file to store the previous list, else update it to the current list.
+     */
+    public void createFileIfNotExists() {
+        Path directoryPath = this.path.getParent();
+
+        if (!Files.exists(directoryPath)) {
+            try {
+                Files.createDirectories(directoryPath);
+                System.out.println("directory created");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        if(!Files.exists(this.path)) {
+            try {
+                Files.createFile(this.path);
+                System.out.println("file created");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Update the list of tasks in the duke.txt according to the current list
+     *
+     * @param tasks the ArrayList of tasks that were added
+     */
+    public void saveTasksToFile(TaskList tasks) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(this.path.toFile()))) {
+            for (int i = 0; i < tasks.size(); i++) {
+                int currentNumber = i + 1;
+                writer.write(tasks.get(currentNumber).toString() + "\n");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Read the list of tasks stored in the duke.txt file and return the ArrayList of tasks.
+     *
+     * @return The ArrayList of tasks saved from previous usage.
+     */
+
+
```

Comment:
According to coding standards, perhaps there should not be empty lines below javadocs and the method declaration?